openedx/core/djangoapps/catalog/utils.py

```diff
@@ -276,12 +276,18 @@ def get_fulfillable_course_runs_for_entitlement(entitlement, course_runs):
     2) A user can enroll in
     3) A user can upgrade in
     4) Are published
+    5) Are not enrolled in already for an active session
 
     These are the only sessions that can be selected for an entitlement.
     """
 
     enrollable_sessions = []
 
+    enrollments_for_user = CourseEnrollment.enrollments_for_user(entitlement.user).filter(mode=CourseMode.VERIFIED)
+    enrolled_sessions = []
+    for course_enrollment in enrollments_for_user:
+        enrolled_sessions.append(str(course_enrollment.course_id))
```

Comment:
This is a matter of taste, but my inner Pythonista would do the above three lines with a list comprehension like:
enrolled_sessions = [str(e.course_id) for e in enrollments_for_user]

Or maybe even better since the only thing you do with this afterwards is test membership:
enrolled_sessions = frozenset([str(e.course_id) for e in enrollments_for_user])

Even if you don't like the list comprehension, using a set() and then add()ing to it might be more performant when you're testing membership later.